testng-core-api/src/main/java/org/testng/IParameterInfo.java

```diff
@@ -0,0 +1,36 @@
+package org.testng;
+
+/** Represents the ability to retrieve the parameters associated with a factory method. */
+public interface IParameterInfo {
```

Comment:
I'm not sure about the name of the public API.
It is related to the factory by doc but not by code.

A factory method produces instances of test classes.
What do you think about something like `TestClassInstance`?

Reply:
Yeah. `TestClassInstance` sounds good to me. I will change this.